test/logflare/backends/dynamic_pipeline_test.exs

```diff
@@ -177,4 +177,64 @@ defmodule Logflare.DynamicPipelineTest do
 
     assert DynamicPipeline.whereis(name) == pid
   end
+
+  test "trace increases and decreases of pipelines",
+       %{name: name, pipeline_args: pipeline_args} do
+    ref_increment =
+      :telemetry_test.attach_event_handlers(self(), [
+        [:logflare, :backends, :dynamic_pipeline, :increment]
+      ])
+
+    ref_decrement =
+      :telemetry_test.attach_event_handlers(self(), [
+        [:logflare, :backends, :dynamic_pipeline, :decrement]
+      ])
```

Comment:
I added two different events for the dynamic pipeline, 

Based on [Backends](https://github.com/Logflare/logflare/blob/main/lib/logflare/backends.ex#L571) we add a telemetry for each event, which what I was trying to replicate here

But I'm unsure if this is the right way to do it since we count those values on `Logflare.Telemetry`